### ISSUE TITLE
Size limit in requests

### ISSUE THREAD

**Issue by ric70x7** [disarm-platform/fn-covariate-extractor#12]
@onlyjsmith 
The function fails to return an output when the request exceeds certain number of points. In this case 2,343. In my tests, 2,342 points is OK, but beyond that we have a problem (json request attached). Any way to sort this out. I was originally trying to make a request with approx. 11,000 points.

Note 1: The issue is not related to the 2,343-th point, since I can make other requests with that point included.
Note 2: The file attached is a .json file, but I changed the extension to .txt, because Github doesn't like the former.
[bad_request.txt](https://github.com/disarm-platform/fn-covariate-extractor/files/3015059/bad_request.txt)

**Comment by HughSt** [comment]
I tried increasing the timeout to 600, but this hasn't resolved the issue.

**Comment by HughSt** [comment]
Update: The function can handle the request fine if run locally or directly invoked on OpenFaas UI, so doesn't seem to be a problem there.

**Comment by ric70x7** [comment]
More info about this: I'm making the call to the function in a Python session, with the command request.post:
- If I use timeout=.1, then the function fails due to the time being exceeded, which is expected. 
- If I use timeout=500, then the function fails but this time sends a message about a possible memory leak.

**Comment by HughSt** [comment]
@ric70x7 did we resolve this?

**Comment by ric70x7** [comment]
No we haven't, yet. There is a problem when we have a very large set of points in the request.
Here is a request with over 61K points.

open_faas_link = 'http://faas.srv.disarm.io/function/fn-covariate-extractor'
req_options ={
                       'points': 'https://www.dropbox.com/s/1x57v4ilyk4jbe6/tha_centroids.json?dl=1',
                       'layer_names': ['elev_m',  'dist_to_water_m',
                                                'bioclim1', 'bioclim4', 'bioclim12', 'bioclim15']
}
covs_response = requests.post(open_faas_link, json=req_options) 

This is the response I get:
{'type': 'error', 'result': 'Error: cannot allocate vector of size 13.9 Gb\n'}

The gejson file stored in dropbox has a size of 8.2MB. I don't know how the request ends up requiring almost 14Gb of space.

**Comment by HughSt** [comment]
@onlyjsmith could you take a look at this issue?

**Comment by HughSt** [comment]
OK - I found the issue. When calculating distance to water I was using an inefficient approach building a complete distance matrix. Have implemented much more efficient method which doesn't require so much memory. Have tested and deployed. This version also accepts a URL which simplifies calling the algo. Closing issue for the moment, but this might crop up again if the function tries to return a very large JSON.
